File: 240726/가장 먼저 나오는 숫자/first-appear-number.js
const main = () => {
    const input = require('fs').readFileSync(0).toString().trim().split("\n");
    const nums = input[1].trim().split(" ").map(Number);
    const answers = input[2].trim().split(" ").map(Number);

    answers.forEach(a => {
        console.log(search(a, nums));
    })
}

const search = (val, nums) => {
    let l = 0;
    let r = nums.length - 1;
    let idx = -1;

    while (l <= r) {
        const mid = Math.floor((l + r) / 2);

        if (nums[mid] === val) idx = mid + 1;

        if (val <= nums[mid]) r = mid - 1;
        else l = mid + 1;
    }

    return idx;
}

main();

File: 240728/가장 먼저 나오는 숫자/first-appear-number.kt
import java.io.*

var arr: Array<Int> = emptyArray()

fun main() {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val nums =
        br
            .readLine()
            .trim()
            .split(" ")
            .map { it.toInt() }
            .toTypedArray()
    arr =
        br
            .readLine()
            .trim()
            .split(" ")
            .map { it.toInt() }
            .toTypedArray()
        br
            .readLine()
            .trim()
            .split(" ")
            .forEach { search(it.toInt()) }
}

fun search(target: Int) {
    var left = 0
    var right = arr.size - 1
    var result = -1
    while (left <= right) {
        val mid = (left + right) / 2
        val num = arr[mid]
        if (num == target) {
            result = mid + 1
            right = mid - 1
        } else if (num > target) {
            right = mid - 1
        } else {
            left = mid + 1
        }
    }
    println(result)
}